src/Extensions/Azure/DurableTaskSchedulerOptions.cs

```diff
@@ -0,0 +1,162 @@
+﻿﻿// Copyright (c) Microsoft Corporation.
+﻿// Licensed under the MIT License.
+
+using System.Globalization;
+using Azure.Core;
+using Azure.Identity;
+
+namespace Microsoft.DurableTask.Extensions.Azure;
+
+/// <summary>
+/// Options for configuring the Durable Task Scheduler.
+/// </summary>
+public class DurableTaskSchedulerOptions
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DurableTaskSchedulerOptions"/> class.
+    /// </summary>
+    internal DurableTaskSchedulerOptions(string endpointAddress, string taskHubName, TokenCredential? credential = null)
```

Comment:
To follow options pattern, this will need a parameterless constructor. You can leverage options validation to ensure all required fields are set.

https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-9.0#options-validation

Reply:
updated